### PR TITLE
feat: implement RBAC for tokens

### PR DESCRIPTION
**Pull request by danut13**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- Implement RBAC for Pantos managed tokens
- Extract RBAC logic for non-diamond contracts and reuse it

### CODE REVIEW

File: src/access/PantosRoles.sol [hunk 1]
```diff
@@ -7,7 +7,7 @@ pragma solidity 0.8.26;
  *
  * @notice Pantos roles defined as bytes32 constants.
  */
-contract PantosRoles {
+library PantosRoles {
```
Comment (chandraprakash) [inline src/access/PantosRoles.sol:10]: Just curious on why you decided to use (embedded) library vs contract(abstract)

**Review by chandraprakash — APPROVED** [review]
Worth adding tests to cover for cross role scenarios like: 
pauser trying to do super critical only actions.

File: src/access/PantosRoles.sol (hunk 1 above)
Comment (danut13) [reply src/access/PantosRoles.sol:10]: I had a discussion with @markuslevonyak and he suggested that logically it's not an "is a" but much rather a "uses" relationship between the contracts which have a relationship with this contract. So a library would be more appropriate

**Comment by danut13** [comment]
@chandraprakash we have tests for non-role actions, things like: `test_unpause_ByNonSuperCriticalOps`. Introducing tests for a number of roles/addresses that try to do actions sounds more like in the domain of fuzz testing to me, and not unit tests

**Review by markuslevonyak — APPROVED** [review]

**Pull request merged**